src/deploy/functions/containerCleaner.ts

```diff
@@ -39,6 +40,25 @@ const SUBDOMAIN_MAPPING: Record<string, string> = {
   "australia-southeast1": "asia",
 };
 
+async function retry<Return>(func: () => Promise<Return>): Promise<Return> {
+  const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+  const MAX_RETRIES = 3;
+  const INITIAL_BACKOFF = 100;
+  let retry = 0;
+  while (true) {
+    try {
+      return await func();
+    } catch (error) {
+      logger.debug("Failed docker command with error", error);
+      retry += 1;
+      if (retry >= MAX_RETRIES) {
+        throw new FirebaseError("Failed to clean up artifacts", { original: error });
```

Comment:
Should we actually consider it an error if we fail to clean up artifacts, or should we print a warning? I'm not sure I'd want to consider it an error, especially since until very recently we didn't clean them up at all...

Reply:
+1 for warning

Reply:
SGTM. I will catch the error higher up and warn.

Reply:
Turns out I already did.